src/backend/distributed/commands/multi_copy.c

```diff
@@ -2118,10 +2136,24 @@ ShouldExecuteCopyLocally(bool isIntermediateResult)
 
 		return true;
 	}
+	else if (IsMultiStatementTransaction())
+	{
+		return true;
+	}
 
-	/* if we connected to the localhost via a connection, we might not be able to see some previous changes that are done via the connection */
-	return GetCurrentLocalExecutionStatus() != LOCAL_EXECUTION_DISABLED &&
-		   IsMultiStatementTransaction();
+	/*
+	 * If we can reserve a connection, we should go ahead with
```

Comment:
I may add one more check such as `if (!RelationHasLocalPlacemen(relationId)) return false`

To mainly prevent some CPU cycles if the coordinator is added to the metadata but shouldhaveshards is false.

Given that if shouldhaveshards=false the shared counters become less relevant other than skipping some code/shared counter waits.

On the other hand, doing that requires loading all the placements of the table upfront, which we seem to not do.

Still, I have a slight preference to add the check.  Do you have a preference?

Reply:
I added for now, if you think otherwise I can rollback this change.